builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -443,6 +441,98 @@ func resourceVSphereVirtualMachineUpdate(d *schema.ResourceData, meta interface{
 	if err != nil {
 		return err
 	}
+
+	if d.HasChange("disk") {
+		hasChanges = true
+		oldDisks, newDisks := d.GetChange("disk")
+		oldDiskSet := oldDisks.(*schema.Set)
+		newDiskSet := newDisks.(*schema.Set)
+
+		addedDisks := newDiskSet.Difference(oldDiskSet)
+		removedDisks := oldDiskSet.Difference(newDiskSet)
+
+		// Removed disks
+		for _, diskRaw := range removedDisks.List() {
+			if disk, ok := diskRaw.(map[string]interface{}); ok {
+				devices, err := vm.Device(context.TODO())
+				if err != nil {
+					log.Printf("[ERROR] Update Remove Disk - Could not get virtual device list")
+				}
+				virtualDisk := devices.FindByKey(int32(disk["key"].(int)))
+
+				keep := false
+				if v, ok := d.GetOk("keep_on_remove"); ok {
+					keep = v.(bool)
+				}
+
+				vm.RemoveDevice(context.TODO(), keep, virtualDisk)
+			}
+		}
+		// Added disks
+		for _, diskRaw := range addedDisks.List() {
+			if disk, ok := diskRaw.(map[string]interface{}); ok {
+
+				var datastore *object.Datastore
+				if disk["datastore"] == "" {
+					datastore, err = finder.DefaultDatastore(context.TODO())
+					if err != nil {
+						return err
+					}
+				} else {
+					datastore, err = finder.Datastore(context.TODO(), disk["datastore"].(string))
+					if err != nil {
+						log.Printf("[ERROR] Couldn't find datastore %v.  %s", disk["datastore"].(string), err)
+						return err
+					}
+				}
+
+				var size int64
+				if disk["size"] == 0 {
+					size = 0
+				} else {
+					size = int64(disk["size"].(int))
+				}
+
+				iops := int64(disk["iops"].(int))
```

Comment:
I believe so.  This is grabbing disk info from the schema, which will always pad with zero-values if the user doesn't not provide a value.  Meaning, for iops, we'd get a 0, not a nil.